src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx

```diff
@@ -1,201 +1,272 @@
-import React, { ChangeEvent, Fragment, useEffect, useState } from 'react'
+import React, { ChangeEvent, useEffect, useMemo, useState } from 'react'
 import Checkbox from 'react-three-state-checkbox'
-import { Form, FormikProps } from 'formik'
 import { observer } from 'mobx-react-lite'
 
 import { FuncStepTypesEnum } from '@core/enum/func-step-types-enum'
 import { t } from '@i18n'
+import datasetStore from '@store/dataset'
 import filterStore from '@store/filter'
+import {
+  ConditionJoinMode,
+  TFuncCondition,
+  TVariant,
+} from '@service-providers/common/common.interface'
+import {
+  IInheritanceModeCachedValues,
+  IInheritanceModeStateConditions,
+} from '../function-panel.interface'
+import functionPanelStore from '../function-panel.store'
 import { PanelButtons } from './panelButtons'
 
-export interface IInheritanceFormValues {
-  problemGroups: string[]
-  variants: string[]
-}
-
-export const InheritanceMode = observer(
-  ({
-    values,
-    setFieldValue,
-    submitForm,
-    resetForm,
-  }: FormikProps<IInheritanceFormValues>) => {
-    const cachedValues =
-      filterStore.readFilterCondition<IInheritanceFormValues>(
+export const InheritanceMode = observer(() => {
+  const cachedValues =
+    filterStore.readFilterCondition<IInheritanceModeCachedValues>(
+      FuncStepTypesEnum.InheritanceMode,
+    )
+
+  const selectedFilterName: string = functionPanelStore.filterName
+  const selectedFilterGroup: string = functionPanelStore.filterGroup
+  const problemGroups: string[] = functionPanelStore.problemGroups
+  const variants: [string, number][] = functionPanelStore.complexVariants
+
+  const filteredVariants: [string, number][] = useMemo(() => {
+    return variants.filter(([, variantValue]) => variantValue > 0)
+  }, [variants])
+
+  // values for requests
+  const [conditions, setConditions] = useState<IInheritanceModeStateConditions>(
```

Comment:
Why do we need this separate state? As i see, it's updated when `cachedVales` changed. And assigned together with `cachedValues` in `handleSetConditions` and other handlers.

May be we can use some derived from `cachedValue`, or `cachedValue` as is?

Reply:
Wow, that's really useful to have such an outside view. It may work without any state indeed!